### PR TITLE
Submitted page

### PR DESCRIPTION
**Pull request by zineanteoh**
Front page
Screen Shot 2022-09-12 at 10 23 02 PM

After logging in for the first time as a hacker
Screen Shot 2022-09-12 at 10 23 26 PM

After submitting the application form
Screen Shot 2022-09-12 at 10 22 39 PM

### CODE REVIEW

**Comment by aadibajpai** [comment]
I think the "Not you" text is not necessary, removing that would allow us to inline the text and the button

**Comment by zineanteoh** [comment]
@aadibajpai ready to be reviewed & merged. 

Fixed: 
- Added file size limit (1 MB)
- Added validation for uploaded pdf (previously when user remove uploaded file they could proceed to submit form)
- Repeating API calls (caused by useSWR revalidatingOnFocus)
- Some stylings (register, submit button, etc.)
- Some other form stuffs (date of birth > 18 years, etc.)

File: pages/HackerDash.tsx [hunk 1]
```diff
@@ -1,27 +1,42 @@
...
 import TeamManager from '../components/TeamManager';
 import TeamSetup from '../components/TeamSetup';
 import { TeamProfile } from '../types/client';
 import { ApplicationStatus, UserData } from '../types/database';
 import styles from '../styles/Form.module.css';
+import { signOut } from 'next-auth/react';
+import moment from 'moment';
 
-export default function HackerDash() {
+type Props = {
```
Comment (aadibajpai) [inline pages/HackerDash.tsx:13]: probably should be more descriptive, HackerProps or sth

File: pages/HackerDash.tsx [hunk 2]
```diff
@@ -1,27 +1,42 @@
...
+	const { data: user } = useSWR(
+		'/api/user-data',
+		async url => {
+			const res = await fetch(url, { method: 'GET' });
+			return (await res.json()) as UserData;
+		},
+		{ revalidateOnFocus: false, revalidateOnMount: true }
+	);
+
+	if (user && user.applicationStatus) {
```
Comment (aadibajpai) [inline pages/HackerDash.tsx:37]:
```suggestion
	if (user?.applicationStatus) {
```

File: pages/HackerDash.tsx [hunk 3]
```diff
@@ -61,6 +85,36 @@ export default function HackerDash() {
...
 
+	function disabledDate(current: moment.Moment): boolean {
+		// disable all dates from 18 years ago onwards
+		return current && current > moment().subtract(18, 'years');
+	}
+
+	const [resumeFile, setResumeFile] = useState<UploadFile[]>([]);
+	const [resumeToUpload, setResumeToUpload] = useState<string | Blob>('');
+
+	const dummyRequest = (options: any) => {
```
Comment (aadibajpai) [inline pages/HackerDash.tsx:96]: why do we need them? should add a comment

File: pages/index.tsx [hunk 4]
```diff
@@ -1,30 +1,40 @@
...
 				<>
-					Signed in as {session.user?.email} <br />
-					<button onClick={() => signOut()}>Sign out</button> <br />
 					<br />
-					{session.userType === 'HACKER' && <HackerDash />}
+					{session.userType === 'HACKER' && (
+						<HackerDash
+							userApplicationStatus={userApplicationStatus}
+							setUserApplicationStatus={setUserApplicationStatus}
+							userEmail={session.user?.email}
```
Comment (aadibajpai) [inline pages/index.tsx:35]: at this juncture why would user not have email?

**Comment by aadibajpai** [comment]
> @aadibajpai ready to be reviewed & merged.
> 
> Fixed:
> 
> * Added file size limit (1 MB)
> * Added validation for uploaded pdf (previously when user remove uploaded file they could proceed to submit form)
> * Repeating API calls (caused by useSWR revalidatingOnFocus)
> * Some stylings (register, submit button, etc.)
> * Some other form stuffs (date of birth > 18 years, etc.)
> 
> Screen Shot 2022-09-14 at 1 13 55 AM
> 
> Screen Shot 2022-09-14 at 1 14 54 AM
> 
> Screen Shot 2022-09-14 at 1 15 22 AM

looks good I left some comments. We should probably bump limit to 10mb or so, to not miss any legit resumes which for some reason are > 1mb

**Comment by zineanteoh** [comment]
Change to a more fitting logo

Screen Shot 2022-09-14 at 1 52 24 AM

File: pages/index.tsx (hunk 4 above)
Comment (zineanteoh) [reply pages/index.tsx:35]: presumably because they haven't logged in yet (happy to change if there's a better approach)

**Comment by aadibajpai** [comment]
> Change to a more fitting logo
> 
> Screen Shot 2022-09-14 at 1 52 24 AM

looks worse imo, we should revert to the previous one. It's not aligned well either

**Review by aadibajpai — APPROVED** [review]

**Pull request merged**